app/src/main.py

```diff
@@ -6,4 +6,12 @@
 @main.route('/profile')
 @login_required
 def profile():
-    return render_template('user_profile.html', name=current_user.name, email=current_user.email)
\ No newline at end of file
+    return render_template(
+        'user_profile.html', 
+        name=current_user.name, 
+        username=current_user.username,
+        email=current_user.email,
+        age=current_user.age,
+        grade=current_user.grade.value,
+        logged_in=True
+    )
```

Comment:
Similar thing here. What if the age or grade wasn't provided? We need some sort of solution to this.

Reply:
As mentioned above, the way the front end page is implemented forces new users to fill out all fields, so there shouldn't be a case where a new user has missing information.